docs/README-METRICS.md

```diff
@@ -0,0 +1,165 @@
+> `MEAN(DAU(d) for E - 7 < d <= E) / MAU(E)`
+
+A daily active user will be defined as any user, using their browser that day,
+with at least one test currently enabled in their browser.  This means only
+having the Test Pilot extension installed will *not* count as an active user.
+
+![](metrics-engagement-ratio.png)
+
+## Engagement Breakdown
+
+Within the above Engagement Ratio of a user having at least one test enabled,
+it is interesting to see how many users have how many tests enabled, including
+the negative space of users with Test Pilot installed but with no tests
+enabled.
+
+Total Users here is defined as a Participating User: a user with the Test Pilot
+add-on installed and has run Firefox that day.  They may or may not have any
+tests enabled.
+
+Below is a mockup of what that could look like.
+
+![](metrics-engagement-breakdown.png)
+
+## Service Health
+
+Test Pilot has two critical functions:  firstly to be the broker enabling and
+disabling tests, and secondly, to collect user feedback.  Our main system
+health measures will be the number of tests installed over time and the count
+of user feedback recorded over time.  These will allow us to quickly see
+disruptions in service regardless of cause.
+
+![](metrics-feedback-recorded.png)
+![](metrics-total-installs.png)
+
+# Data Collection Details
+
+There are two sources for metrics data - the server and the client.
+
+**The server** will use server logs to store metric data which will be picked
+up by the measurement team ([some docs][3]), parsed and transformed if
+necessary, and pushed into ElasticSearch (as scale permits), S3, and Redshift.
+Events recorded in the logs will include:
+
+* New user registration
+* New feedback about a test received
+* Test Pilot add-on installed
+* Test installed
+
+Data recorded by the log files will include standard system logging as well as:
+* Test Pilot User ID (if applicable)
+* Test ID (if applicable)
+* Feature Switch Set (if applicable)
+* User Agent
+* Campaign (marketing identifier) (if applicable)
+
+Using the [standard format][4] an example object will look like:
+
+```json
+{
+  "Timestamp": 1457462527,
+  "Type": "request.summary",
+  "Logger": "TestPilotWeb",
+  "Hostname": "testpilot.servers.s3",
+  "EnvVersion": "2.0",  // We're conforming to the Fields spec
+  "Severity": 0,
+  "Pid": 1000,
+  "Fields":{
+    "agent": "User Agent String",
+    "path": "/tests/",
+    "method": "GET",
+    "code": 200,
+    "errno": 0,
+    "lang": "en-US",
+    "uid": 1000,  // This is the Test Pilot UID
+    "service": "",
+    "context": "",
+    "msg": "New Test Installed: Universal Search",
+    "remoteAddressChain": ["1.2.3.4", "4.3.2.1"],
+    "rid": "0b65c6fb-26d5-4367-be29-b3094f0c69d5",
+    "t": 256,
+
+    "test_id": "universal_search",
+    "feature_switches": "",
+    "campaign": ""
+  }
+}
+```
+
+The second source, **the client**, will use the existing Telemetry system (the
```

Comment:
Still wrapping my head around how Telemetry works... I think we also want to declare / define our own "testpilot" ping that extends from the common format [like these other ones](https://gecko.readthedocs.org/en/latest/toolkit/components/telemetry/telemetry/index.html).